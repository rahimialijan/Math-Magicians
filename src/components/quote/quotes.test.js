import React from 'react';
import { render, waitFor } from '@testing-library/react';
import Quotes from './quotes';

describe('Quotes component', () => {
  it('should render loading state when data is loading', () => {
    const { asFragment } = render(<Quotes />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error state when data cannot be fetched', async () => {
    // Mock the fetch request to simulate an error
    global.fetch = jest.fn(() => Promise.reject(new Error('Failed to fetch data')));

    const { asFragment } = render(<Quotes />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should render the quote correctly when data is fetched', async () => {
    // Mock the successful fetch request and provide sample data
    const mockData = {
      category: 'Design',
      author: 'John Doe',
      quote: 'Sample quote',
    };
    global.fetch = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve([mockData]),
    }));

    const { asFragment, getByText } = render(<Quotes />);
    await waitFor(() => {
      expect(getByText('Category:')).toBeInTheDocument();
      expect(getByText(mockData.category)).toBeInTheDocument();
      expect(getByText('Author:')).toBeInTheDocument();
      expect(getByText(mockData.author)).toBeInTheDocument();
      expect(getByText('Quote:')).toBeInTheDocument();
      expect(getByText(mockData.quote)).toBeInTheDocument();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
