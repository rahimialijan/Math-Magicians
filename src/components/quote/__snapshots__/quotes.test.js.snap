// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component should render error state when data cannot be fetched 1`] = `
<DocumentFragment>
  <div>
    Page is Loading...
  </div>
</DocumentFragment>
`;

exports[`Quotes component should render loading state when data is loading 1`] = `
<DocumentFragment>
  <div>
    Page is Loading...
  </div>
</DocumentFragment>
`;

exports[`Quotes component should render the quote correctly when data is fetched 1`] = `
<DocumentFragment>
  <div
    class="quotes-container"
  >
    <div
      class="quotes-wrapper"
    >
      <h3>
         
        <strong>
          Category:
        </strong>
         Design
      </h3>
      <span>
        <strong>
          Author:
        </strong>
         John Doe
      </span>
      <p>
        <strong>
          Quote:
        </strong>
         Sample quote
      </p>
    </div>
  </div>
</DocumentFragment>
`;
